src/posts/2020-07-27-unsucking-your-project-onboarding.md

```diff
@@ -0,0 +1,789 @@
+. "./.envrc"
+
+pp_info "test" "running frontend tests"
+yarn jest
+
+pp_info "test" "running elixir tests"
+mix test
+
+pp_success "test" "tests passed!"
+```
+
+### bin/client
+
+When the backend and frontend are run separately, `bin/server` becomes
+responsible for running the server only and we add in a `bin/client`.
+
+As an example, this is a Node script that runs `webpack-dev-server` separately:
+
+```javascript
+#!/usr/bin/env node
+
+/* eslint-disable no-console */
+
+const webpack = require("webpack");
+const WebpackDevServer = require("webpack-dev-server");
+const config = require("../webpack.config");
+
+new WebpackDevServer(webpack(config), config.devServer).listen(
+  config.devServer.port,
+  config.devServer.host,
+  err => {
+    if (err) console.error(err);
+  }
+);
+
+process.stdin.resume();
+process.stdin.on("end", () => {
+  process.exit(0);
+});
+```
+
+### bin/ci
+
+Sometimes found as an abstraction to help developers mimic the CI behaviour.
+Usually comprised by `bin/test` and `bin/lint`.
+
+```shell
+#!/usr/bin/env sh
+
+set -e
+
+. "./bin/functions"
+
+pp_info "ci" "running local ci script"
+
+./bin/lint
+./bin/test
+
+pp_success "ci" "all good"
+```
+
+### bin/ci-helpers
+
+When the CI has a complex setup, it's usually easier to abstract these into
+helper scripts and call them directly. We usually don't call these scripts
+locally, instead they're only used by the CI.
+
+This script would manage Elixir caching on Semaphore:
+
+```shell
+#!/usr/bin/env sh
+
+set -e
+
+ELIXIR_VERSION=$(cat .tool-versions | grep elixir | cut -d ' ' -f 2)
+ELIXIR_CACHE_KEY=kiex-elixir-$ELIXIR_VERSION
+
+if cache has_key "$ELIXIR_CACHE_KEY"; then
+  cache restore $ELIXIR_CACHE_KEY
+else
+  kiex install $ELIXIR_VERSION
+  cache store $ELIXIR_CACHE_KEY $HOME/.kiex/elixirs
+fi
+
+kiex default $ELIXIR_VERSION
+. $HOME/.kiex/elixirs/elixir-$ELIXIR_VERSION.env
+
+mix local.hex --force
+mix local.rebar --force
+```
+
+### bin/console
+
+Sometimes we need special instructions to connect to a production or staging
+server or even locally, when there are complex steps to connect to the console.
+
+### bin/docs
+
+Simple scripts to call to generate docs. Usually just to abstract the doc
+generation process so that newcommers don't need to know the intricacies of each
+language of framework.
+
+## Into hyperdrive
+
+As much as we aim to simplify and abstract the process of joining a new project,
+several of us take this to a different level: abstracting the process of getting
+a new computer. **The philosophy is that your tool should be expendable. Having to
+set up a new one should be a matter of minutes or, at most, a couple of hours.
+Not days.**
+
+Personally, [I use an install script][dotfiles-install], 100% made in `zsh`. It
+installs and sets up [my dotfiles][my-dotfiles] and the apps I need.
+
+[Gabriel][gabriel-dotfiles] has a different philosophy, going with Ansible.
+[Naps][naps-terraform] uses Terraform and [Pedro][pedro-terraform] made a
```

Comment:
I'd prefer `Miguel` here to be honest

Reply:
and what the hell? I don't use terraform (where did that idea even come from?)
I'm old school and simple, I still use `rcup`